### PR TITLE
GEOMESA-2803  Upgrade to GeoTools 23.0

### PR DESCRIPTION
**Pull request by jnh5y**
Signed-off-by: Jim Hughes <jnh5y@ccri.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
@jnh5y i went through and built it with gt 21.1, then switched the version to 23-RC and just ran surefire:test - i think that guarantees binary compatibility between versions, so hopefully this is good to go now.

File: build/cqs.tsv [hunk 1]
```diff
@@ -74,38 +74,38 @@ io.netty:netty-all	4.1.48.Final	compile
...
+it.geosolutions.jaiext.rlookup:jt-rlookup	1.1.14	compile
+it.geosolutions.jaiext.scale2:jt-scale2	1.1.14	compile
+it.geosolutions.jaiext.scale:jt-scale	1.1.14	compile
+it.geosolutions.jaiext.shadedrelief:jt-shadedrelief	1.1.14	compile
+it.geosolutions.jaiext.stats:jt-stats	1.1.14	compile
+it.geosolutions.jaiext.translate:jt-translate	1.1.14	compile
+it.geosolutions.jaiext.utilities:jt-utilities	1.1.14	compile
+it.geosolutions.jaiext.vectorbin:jt-vectorbin	1.1.14	compile
+it.geosolutions.jaiext.warp:jt-warp	1.1.14	compile
+it.geosolutions.jaiext.zonal:jt-zonal	1.1.14	compile
```
Comment (jnh5y) [inline build/cqs.tsv:108]: http://dev.eclipse.org/ipzilla/show_bug.cgi?id=21925

File: build/cqs.tsv [hunk 2]
```diff
@@ -185,37 +185,36 @@ org.eclipse.xsd:org.eclipse.xsd	2.12.0	compile
...
+org.geotools.ogc:net.opengis.ows	23.0	compile
+org.geotools.ogc:net.opengis.wfs	23.0	compile
+org.geotools.ogc:org.w3.xlink	23.0	compile
+org.geotools.xsd:gt-xsd-core	23.0	compile
+org.geotools.xsd:gt-xsd-fes	23.0	compile
+org.geotools.xsd:gt-xsd-filter	23.0	compile
+org.geotools.xsd:gt-xsd-gml2	23.0	compile
+org.geotools.xsd:gt-xsd-gml3	23.0	compile
+org.geotools.xsd:gt-xsd-ows	23.0	compile
+org.geotools.xsd:gt-xsd-wfs	23.0	compile
```
Comment (jnh5y) [inline build/cqs.tsv:215]: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=21926

File: build/cqs.tsv (hunk 1 above)
Comment (jnh5y) [reply build/cqs.tsv:108]: Approved!

**Comment by elahrvivaz** [comment]
merged as 5ee94656d7806d08d5ca647dfc10685fd230b402

**Pull request closed without merging**